geode-core/src/main/java/org/apache/geode/distributed/internal/membership/InternalDistributedMember.java

```diff
@@ -637,7 +639,51 @@ public boolean equals(Object obj) {
     if ((obj == null) || !(obj instanceof InternalDistributedMember)) {
       return false;
     }
-    return compareTo((InternalDistributedMember) obj) == 0;
+    InternalDistributedMember other = (InternalDistributedMember) obj;
+
+    int myPort = getPort();
+    int otherPort = other.getPort();
+    if (myPort != otherPort) {
+      return false;
+    }
+
+    InetAddress myAddr = getInetAddress();
+    InetAddress otherAddr = other.getInetAddress();
+    if (myAddr == null && otherAddr == null) {
+      return true;
+    } else if (!Objects.equals(myAddr, otherAddr)) {
```

Comment:
Object.equals handles the `null` case as well.

Reply:
I don't think so. Objects.equals(null, null) returns true. But so does other non-null InetAddresses. But we are only willing to stop here if either they are not equal OR they are both null. If the addresses are non-null and equal then we continue on and compare other things. This logic was derived from the compareTo logic that I did not want to change. I'm not sure why it did not treat "null, null" as equal and just continue to compare other things.

Reply:
Oh I see... yeah that is odd.